### ISSUE TITLE
some questions about the data_loss

### ISSUE THREAD

**Issue by hanlu-zju** [voxelmorph/voxelmorph#301]
Dear author, 
Recently, I have trained my model to register CT lung images using legacy branch.
It works well when the data_loss is ‘mse’， but spatial_transformer_1_loss is strange(**eg.  -25284398.000**) in the case of ‘ncc’.

**Details of experiments**
Data's pixel value ranges from -1000 to 1000
The data is not normalized

**What have I tried**
I have tried to train a model with only one moving data(mov_vol) and only one fixed data(atlas_vol), and save the output of 'spatial_transformer_1' as spatial_transformer_output. Another call of NCC().loss with input[atlas_vol, spatial_transformer_1] will give a value whose **range is [-1, 0]**

**My question**
1. In  legacy branch，I can find losses.NCC().loss，but I can't find where 'mse' is defined. 
2. Since I have checked that the calculation of NCC().loss is correct(**range [-1, 0]**), I'm wondering  what are passed to NCC().loss result in strange spatial_transformer_1_loss (**eg.  -25284398.000**)  during keras training. Is there any other relationship between 'spatial_transformer_1_loss' and 'NCC().loss'? How can I find which variable is y_true and which variable is y_predict?
3. Must the data be normalized to [0,1]? After the data is normalized, spatial_transformer_1_loss will give a value whose **range is [-1, 0]** during keras training.
    
```
        # compile
        mg_model.compile(optimizer=Adam(lr=lr), 
                         loss=[data_loss, losses.Grad('l2').loss],
                         loss_weights=[1.0, reg_param])
            
        # fit
        mg_model.fit_generator(cvpr2018_gen, 
                               initial_epoch=initial_epoch,
                               epochs=nb_epochs,
                               callbacks=[save_callback, TensorBoard(log_dir='./log')],
                               steps_per_epoch=steps_per_epoch,
                               verbose=1)
        #  my code to save the output of 'spatial_transformer_1' 
        moved_layer_model = Model(inputs=mg_model.input, outputs=mg_model.get_layer('spatial_transformer_1').output)
        mov_vol = np.fromfile(train_vol_names[0], dtype=np.float32)
        mov_vol = mov_vol.reshape(128, 128, 128)[np.newaxis, ..., np.newaxis]
        np.save('mov_vol.npy', mov_vol)
        np.save('atlas_vol.npy', atlas_vol)
        spatial_transformer_output = moved_layer_model.predict([mov_vol, atlas_vol])
        np.save('spatial_transformer_output.npy', spatial_transformer_output)
```

```
fix = np.load('atlas_vol.npy')
moved = np.load('spatial_transformer_output.npy')
fix_tensor = tf.convert_to_tensor(fix, dtype=tf.float32)
moved_tensor = tf.convert_to_tensor(moved, dtype=tf.float32)
loss_ncc = NCC().loss(fix_tensor, moved_tensor)

with tf.Session() as sess1:
    print(sess1.run(NCC().loss(fix_tensor, moved_tensor)))
```

**Comment by adalca** [comment]
@hanlu-zju since you closed this, does this mean you figured it out?
